### PR TITLE
Add macro helper to compute LaneSwizzle offset

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
Added a macro to help on the creation of the offset that the lane swizzle intrinsic requires.

### CODE REVIEW

**Comment by EvgeniiG** [comment]
Could you test this on the AABB PR, please?
Also, could you perhaps also expose LaneSwizzle(v, andMask, orMask, xorMask) as a macro? Note: only for XBox.
Thanks!

**Comment by FrancescoC-unity** [comment]
> Could you test this on the AABB PR, please?
> Also, could you perhaps also expose LaneSwizzle(v, andMask, orMask, xorMask) as a macro? Note: only for XBox.
> Thanks!

Sure, if you make the loops into macros I'll give it a shot tomorrow morning.

**Comment by EvgeniiG** [comment]
> > Could you test this on the AABB PR, please?
> > Also, could you perhaps also expose LaneSwizzle(v, andMask, orMask, xorMask) as a macro? Note: only for XBox.
> > Thanks!
> 
> Sure, if you make the loops into macros I'll give it a shot tomorrow morning.

If the existing loop with constant parameters doesn't work I may as well just write a specific version for Xbox...

**Comment by FrancescoC-unity** [comment]
> > > Could you test this on the AABB PR, please?
> > > Also, could you perhaps also expose LaneSwizzle(v, andMask, orMask, xorMask) as a macro? Note: only for XBox.
> > > Thanks!
> > 
> > 
> > Sure, if you make the loops into macros I'll give it a shot tomorrow morning.
> 
> If the existing loop with constant parameters doesn't work I may as well just write a specific version for Xbox...

As is now even with the macro it doesn't work since it is not using literals (i.e. the xorMask  is using the index of the loop, need explicit unrolling)

**Review by JulienIgnace-Unity — APPROVED** [review]

File: com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl [hunk 1]
```diff
@@ -215,6 +215,12 @@
 #define WaveGetLaneCount ERROR_ON_UNSUPPORTED_FUNCTION(WaveGetLaneCount)
 #endif
 
+#if defined(PLATFORM_SUPPORTS_WAVE_INTRINSICS)
+// Helper macro to compute lane swizzle offset starting from andMask, orMask and xorMask.
+// IMPORTANT, to guarantee compatibility with all platforms, the masks need to be constant literals (constants at compile time)
+#define LANE_SWIZZLE_OFFSET(andMask, orMask, xorMask)  andMask | (orMask << 5) | (xorMask << 10)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl:221]: add ( )

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**